### PR TITLE
CSI: design code such that volume changes are possible

### PR DESCRIPTION
**Pull request by amarts**
* add 'inotify' to watch for changes for volfile
* handle SIGHUP in python script to regenerate volfile, and send SIGHUP command
* as a backup check configmap file of the volume everytime any management
  operation using 'mount' is made.

Signed-off-by: Amar Tumballi <amar@kadalu.io>

### CODE REVIEW

File: csi/main.py [hunk 1]
```diff
@@ -42,6 +44,23 @@ def mount_storage():
...
+    Reconfigure the mounts by regenerating the volfiles.
+    """
+    host_volumes = get_pv_hosting_volumes({})
+    for volume in host_volumes:
+        if volume["type"] == "External":
+            # Need to skip remount external
+            continue
+        hvol = volume["name"]
+        try:
+            if reload_glusterfs(volume):
```
Comment (leelavg) [inline csi/main.py]: Exception (cmdexception) is already handled in `reload_glusterfs` without re-raising it. I guess just `if` and `else` would suffice without exception handling?

**Review by leelavg — COMMENTED** [review]
Minor comment

**Comment by amarts** [comment]
This is one of the commits from #555 

Needed https://github.com/kadalu/glusterfs/commit/aec4ba60b2f264ea7190a05aa12fe14e0355d846 for properly handling SIGHUP in case of kadalu volumes.

**Review by leelavg — APPROVED** [review]
LGTM

**Pull request merged**